preview/Win7Msix/Win7MSIXInstaller/StartupTask.cpp

```diff
@@ -0,0 +1,193 @@
+#include <windows.h>
+
+#include <shlobj_core.h>
+#include <CommCtrl.h>
+
+#include "FilePaths.hpp"
+#include "StartupTask.hpp"
+#include "GeneralUtil.hpp"
+#include "Constants.hpp"
+#include <TraceLoggingProvider.h>
+#include <taskschd.h>
+
+const PCWSTR StartupTask::HandlerName = L"StartupTask";
+
+const PCWSTR StartupTask::TaskDefinitionXmlPrefix =

```

Comment:
I am assuming that this definition is compatible with what ships in Win10. May be add a comment to that effect.  Also are there any settings that can be influenced based on the xml in the manifest?

Reply:
this is NOT compatible with win10 objects, as win10 has OS support for specifically allowing apps to enable/disable their startup tasks, and OS support to run the tasks as part of the logon sequence. This instead creates tasks using task scheduler because there is no other OS support in win7.

---
In reply to: [277071361](https://github.com/Microsoft/msix-packaging/pull/87#discussion_r277071361) [](ancestors = 277071361)

Reply:
updated the comment in .hpp to make this clear

---
In reply to: [277906358](https://github.com/Microsoft/msix-packaging/pull/87#discussion_r277906358) [](ancestors = 277906358,277071361)